src/amalgam/gen/neondot.c

```diff
@@ -5958,3 +5962,160 @@ void xnn_qu8_igemm_minmax_rndnu_ukernel_4x8c4__neondot(
     }
   } while (nc != 0);
 }
+
+void kai_matmul_clamp_f32_qai8dxp1x8_qsu4cxp8x8_1x8x32_neon_dotprod(
+    size_t m,
+    size_t n,
+    size_t k,
+    const int8_t* restrict lhs_p,
+    size_t lhs_stride,
+    const void* restrict rhs_p,
+    float* restrict dst,
+    size_t dst_stride_row,
+    size_t dst_stride_col,
+    const union xnn_f32_qc4w_minmax_params params[restrict XNN_MIN_ELEMENTS(1)],
+    const struct xnn_qd8_quantization_params quantization_params[restrict XNN_MIN_ELEMENTS(1)]) XNN_OOB_READS
+{
+  const size_t kai_mr = 1;
+  const size_t kai_nr = 8;
+  const size_t kai_k0 = 32;
+
+  const float scalar_min = params->scalar.min;
+  const float scalar_max = params->scalar.max;
+
+  if (m == 0) {
```

Comment:
m == 0 should never happen... you can assert.  Its also good to assert all values are in the expected range.